diff --git a/node_modules/ts-paths-esm-loader/resolverFactory.js b/node_modules/ts-paths-esm-loader/resolverFactory.js
index 33f3420..959b978 100644
--- a/node_modules/ts-paths-esm-loader/resolverFactory.js
+++ b/node_modules/ts-paths-esm-loader/resolverFactory.js
@@ -3,10 +3,12 @@ import { pathToFileURL } from 'url'
 import { matchPath } from './matchPath.js'
 
 export const resolverFactory = (resolveTs) => {
-    return function (specifier, ctx, defaultResolve) {
-        const match = matchPath && matchPath(specifier)
-        return match
-          ? resolveTs(pathToFileURL(`${match}`).href, ctx, defaultResolve)
-          : resolveTs(specifier, ctx, defaultResolve)
-      }
-}
+  return function (specifier, ctx, defaultResolve) {
+    if (specifier.endsWith('.js')) {
+      const trimmed = specifier.substring(0, specifier.length - 3)
+      const match = matchPath(trimmed)
+      if (match) return resolveTs(pathToFileURL(`${match}.js`).href, ctx, defaultResolve)
+    }
+    return resolveTs(specifier, ctx, defaultResolve)
+  }
+};
